File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/backend/connected/aas/submodelelement/ConnectedSubmodelElement.java
package org.eclipse.basyx.aas.backend.connected.aas.submodelelement;

import org.eclipse.basyx.aas.api.metamodel.aas.submodelelement.ISubmodelElement;
import org.eclipse.basyx.aas.backend.connected.ConnectedElement;
import org.eclipse.basyx.vab.core.proxy.VABElementProxy;
/**
 * "Connected" implementation of SubmodelElement
 * @author rajashek
 *
 */
public abstract class ConnectedSubmodelElement extends ConnectedElement implements ISubmodelElement {
	public ConnectedSubmodelElement(VABElementProxy proxy) {
		super(proxy);		
	}
}

File: sdks/c++/basys.sdk.cc/src/aas/aas/submodelelement/connected/ConnectedSubmodelElement.cpp
/*
 * ConnectedSubmodelElement.cpp
 *
 *      Author: wendel
 */

#include "ConnectedSubmodelElement.h"
#include "aas/backend/connected/ConnectedElement.h"

basyx::aas::submodelelement::connected::ConnectedSubmodelElement::ConnectedSubmodelElement(std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  backend::ConnectedElement(backend::ConnectedElement(proxy))
{}
